internal/messagequeue/messagequeue.go

```diff
@@ -566,6 +566,7 @@ func (mq *MessageQueue) extractOutgoingMessage(supportsHave bool) (bsmsg.BitSwap
 	}
 
 	// Add each regular want-have / want-block to the message
+	peerSent := make([]wantlist.Entry, 0, len(peerEntries))
```

Comment:
nit: we can actually reuse the original slice if we do this correctly. But this is a low priority as I'm pretty sure we're going to delete this code when we land #347.